architecture/core_concepts/pods_and_services.adoc

```diff
@@ -160,6 +161,61 @@ link:https://github.com/GoogleCloudPlatform/kubernetes/blob/master/docs/user-gui
 has details about the functionality and purpose of pods.
 ====
 
+[[init-containers]]
+== Init Containers
+
+An link:https://kubernetes.io/docs/concepts/workloads/pods/init-containers/[init container] is a container in a pod that is started before the pod app containers are started. Init containers can share volumes, perform network operations, and perform computations before the remaining containers start. Init containers can also block or delay the startup of application containers until some precondition is met. 
+
+When a pod starts, after the network and volumes are initialized, the init containers are started in order. Each init container must exit successfully before the next is invoked. If an init container fails to start (due to the runtime) or exits with failure, it is retried according to the pod 
+xref:../../dev_guide/configmaps.adoc#consuming-configmap-in-pods[`restartPolicy`]: 
+
+* Pods configured with `restartPolicy` set to `Never` immediately fail and exit. 
+* Pods configured with `restartPolicy` set to `Always` retry the failing init container with increasing backoff until it succeeds.
```

Comment:
I combined the two lists into a single list:
* `restartPolicy: Always` tries retarting infinitely, with an exponential back-off delay of (10s, 20s, 40s) until the pod is restarted. 
* `restartPolicy: Never` does not try to restart. Pods immediately fail and exit.  
* `restartPolicy: OnFailure` tries restarting with an exponential back-off delay of (10s, 20s, 40s) capped at five minutes.

The upstream Kubernetes doc suggests that OnFailure does not retry until it succeeds. Is that true for OpenShift?